lux/utils/debug_utils.py

```diff
@@ -22,5 +29,94 @@ def show_versions():
     print(df.to_string(index=False))
 
 
+def notebook_enabled() -> tp.Tuple[bool, str]:
+    status, nbextension_list = subprocess.getstatusoutput("jupyter nbextension list")
+
+    if status != 0:
+        return False, "Failed to run 'jupyter nbextension list'"
+
+    match = re.findall(r"config dir:(.*)\n", nbextension_list)
+
+    if match:
+        config_dir = match[0].strip()
+    else:
+        return False, "No 'config dir' found in 'jupyter nbextension list'"
+
+    notebook_json = Path(config_dir) / "notebook.json"
+
+    if not notebook_json.exists():
+        return False, f"'{notebook_json}' does not exist"
+
+    extensions = json.loads(notebook_json.read_text())
+
+    if "load_extensions" not in extensions:
+        return False, "'load_extensions' not in 'notebook.json'"
+    elif "luxwidget/extension" not in extensions["load_extensions"]:
+        return False, "'luxwidget/extension' not in 'load_extensions'"
+
+    extension_enabled = extensions["load_extensions"]["luxwidget/extension"]
+
+    if not extension_enabled:
+        return False, "luxwidget is installed but not enabled"
+
+    return True, ""
+
+
+def lab_enabled() -> tp.Tuple[bool, str]:
+    status_str, lab_list = subprocess.getstatusoutput("jupyter labextension list")
+
+    if status_str != 0:
+        return False, "Failed to run 'jupyter labextension list'"
+
+    match = re.findall(r"luxwidget (\S+) (\S+) (\S+)", lab_list)
+
+    if match:
+        version_str, enabled_str, status_str = (_strip_ansi(s) for s in match[0])
+    else:
+        return False, "'luxwidget' not found in 'jupyter labextension list'"
+
+    if enabled_str != "enabled":
+        enabled_str = re.sub(r"\033\[(\d|;)+?m", "", enabled_str)
+        return False, f"luxwidget is installed but currently '{enabled_str}'"
+
+    if status_str != "OK":
+        return False, f"luxwidget is installed but currently '{status_str}'"
+
+    return True, ""
+
+
+def is_lab_notebook():
+    import re
+    import psutil
```

Comment:
Note that we need to add psutil on requirements.txt. It seems like psutil is not something that comes as a default installation.

![Screen Shot 2022-01-30 at 10 03 49 AM](https://user-images.githubusercontent.com/5554675/151711593-41360a81-770f-4f4c-a153-67519aa629aa.png)

Reply:
Ah interesting, thanks! It was somehow on my env. I'll add it.